### PR TITLE
post employee optimisation

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: apps/sage_intacct/utils.py [hunk 1]
```diff
@@ -325,18 +325,46 @@ def sync_dimensions(self):
...
+                'email': email
             }
         }, self.workspace_id)
 
         return vendor_attribute
+    
+    def get_or_create_employee(self, source_employee: ExpenseAttribute):
+        """
+        Call Sage Intacct api to get or create employee
+        :param vendor_name: Name of the employee
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:344]: remove

File: apps/sage_intacct/utils.py [hunk 2]
```diff
@@ -325,18 +325,46 @@ def sync_dimensions(self):
...
+        :param email: Email of the employee
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:345]: update these lines

File: apps/sage_intacct/utils.py [hunk 3]
```diff
@@ -325,18 +325,46 @@ def sync_dimensions(self):
...
+        employee_name = source_employee.detail['full_name']
+        employee = self.connection.employees.get(field='CONTACT_NAME', value=employee_name)
+
+        if 'employee' in employee:
+            employee = employee['employee'][0] if int(employee['@totalcount']) > 1 else employee['employee']
+        else:
+            employee = None
+
+        if not employee:
+            created_employee = self.post_employees(employee)
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:358]: ```post_employees()``` expects an `ExpenseAttribute` object, but you are sending None incase if there are no matches? How will this work?

File: apps/sage_intacct/utils.py [hunk 4]
```diff
@@ -325,18 +325,46 @@ def sync_dimensions(self):
...
+    def get_or_create_employee(self, source_employee: ExpenseAttribute):
+        """
+        Call Sage Intacct api to get or create employee
+        :param vendor_name: Name of the employee
+        :param email: Email of the employee
+        :param create: False to just Get and True to Get or Create if not exists
+        :return: Employee
+        """
+        employee_name = source_employee.detail['full_name']
+        employee = self.connection.employees.get(field='CONTACT_NAME', value=employee_name)
```
[the remark this answers is not in the document]
Comment (NileshPant1999) [reply apps/sage_intacct/utils.py:350]: nope without contact no employee exists it is a mandatory field you can check it below
https://www-p02.intacct.com/ia/acct/frameset.phtml?.sess=pCFx5UPla0LFnFPt9aDp16omQsScNg..

**Review by ashwin1111 — APPROVED** [review]

File: apps/sage_intacct/utils.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Shwetabhk) [reply apps/sage_intacct/utils.py:348]: I think Contact name is just a field like `NAME` in Vendor
This should be fine

**Pull request merged**